src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -10,54 +10,112 @@ import { imageUrls } from '../images';
     if (Number.isNaN(index) || index >= imageUrls.length || index < 0) {    
       console.log('Index: ' + index + 'Not a number or greater than number of images, returning to Gallery');
       this.router.navigate(['/gallery']);
     }
   }
-  
-  // Draws the image user selects from gallery on Canvas
-  // Currently, editor can require user to scroll to access
-  //  the entire photo, need to make it so the editor is 
-  //  fixed to screen size
-  private draw() {
+  /**  
+   * Draws the image user selects from gallery on Canvas
+   *    and creates a hidden canvas to store the original image 
+   *    as a reference when scaling the imageUI
+   * TODO(shmcaffrey): Currently, editor can require user to 
+   * scroll to access the entire photo, need to make it 
+   * so the editor is fixed to screen size.
+   */
+  private draw(): void {
     this.image.src = this.url;
 
-    //adjust canvas to image width and height
-    this.canvas.nativeElement.width = this.image.width;
-    this.canvas.nativeElement.height = this.image.height;
+    let imgWidth = this.image.width;
+    let imgHeight = this.image.height;
+
+    // Initialize hidden canvas width and height to original images width and height.
+    this.hiddenCanvas.nativeElement.width = imgWidth;
+    this.hiddenCanvas.nativeElement.height = imgHeight;
+
+    // Used to scale the image to the window size, 1.2 so the image is smaller than the window.
+    this.scaleFactor = Math.floor(this.innerHeight / imgHeight * this.scaleFactor);
+    // TODO(shmcaffrey): add scaling if image is larger than window
+    if (this.scaleFactor <= 0) {
+      this.scaleFactor =  1;
+    }
+
+    // Adjust canvas to scaled image width and height, use ctx. 
+    this.canvas.nativeElement.width = imgWidth* this.scaleFactor;
+    this.canvas.nativeElement.height = imgHeight * this.scaleFactor;
+    this.ctx.scale(this.scaleFactor, this.scaleFactor); 
+
+    //  Initalize transparent black image data to use for mask.
+    this.maskData = new ImageData(imgWidth,  imgHeight);
     
     this.image.onload = () => {
-      this.ctx.drawImage(this.image, 0, 0);
+      this.ctx.drawImage(this.image, 0, 0, imgWidth, imgHeight);
+      this.hiddenCtx.drawImage(this.image, 0, 0)
     }
   }
+
+  /** Returns the calculated scale for the image loaded. */
+  public getScaleFactor(): number {
+    return this.scaleFactor;
+  }
+
+  /** Returns the original images data for reference in mask making. */
+  public getImgData(): ImageData {
+    return this.hiddenCtx.getImageData(0,0, this.image.width, this.image.height);
+  }
+
+  /** Returns black transparent ImageData for single mask image. */
+  public getMaskData(): ImageData {
+    return this.maskData;
+  }
+
+  /** Returns the scaled images data for reference printing scaled image after mask. */
+  public getScaledData(): ImageData {
+    return this.ctx.getImageData(0, 0, this.image.width * this.scaleFactor, this.image.height * this.scaleFactor);
+  }
 }
```

Comment:
I might be missing something but doesn't this return the scaled original image data PLUS the scaled mask?  Since we are getting the image data from the main canvas?  I think it could also be the cumulative mask from multiple user clicks.

If this is correct, then I think the name originalData, scaledData and maskData are a bit confusing.  I assumed scaledData was the originalData just scaled to the size of the canvas.  

I think we should clarify the names a bit. Data is a bit of a generic word.  If possible leave it out.  One suggestion is (but it may not be perfect):

originalData -> originalImage
scaledData - > scaledImageAndMask
maskData -> mask

This function name is probably better called `getScaledImgAndMask`.

Reply:
the scaled mask is an object of type imageData but it's not drawn on any canvas, and if it was it would be transparent and black so it wouldn't be seen. And yes scaledData is just originalData but scaled to the size of the canvas. 

If it's okay, I will update the names for the next pr because I forgot to make a separate branch for the toolbar and I don't want to prolong the pr by adding more files to it which would be necessary to push the name changes and have the pr in working condition.

Reply:
Sounds good to me.  You can resolve the conversation so it doesn't block you.

We should also chat over VC because I am still confused about what getImageData returns.